UMLEditor/UMLEditor/Views/AStarSharp.cs

```diff
@@ -0,0 +1,218 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+
+namespace AStarSharp
+{
+    /// <summary>
+    /// Class to represent a node in the grid
+    /// </summary>
+    public class Node
+    {
+        // Change this depending on what the desired size is for each element in the grid
+        public static int NODE_SIZE = 15;
+        
+        /// <summary>
+        /// Parent node to this node
+        /// </summary>
+        public Node Parent;
+        /// <summary>
+        /// Grid position of the node
+        /// </summary>
+        public Vector2 Position;
+        /// <summary>
+        /// Center position of the node
+        /// </summary>
+        public Vector2 Center
+        {
+            get
+            {
+                return new Vector2(Position.X + NODE_SIZE / 2, Position.Y + NODE_SIZE / 2);
+            }
+        }
+        
+        /// <summary>
+        /// Distance to current target
+        /// </summary>
+        public float DistanceToTarget;
+        /// <summary>
+        /// Cost of reaching this node from current start
+        /// </summary>
+        public float Cost;
+        /// <summary>
+        /// Weight of traversing this node
+        /// </summary>
+        public float Weight;
+        /// <summary>
+        /// Sum of distance and cost
+        /// </summary>
+        public float F
+        {
+            get
+            {
+                if (DistanceToTarget != -1 && Cost != -1)
+                    return DistanceToTarget + Cost;
+                else
+                    return -1;
+            }
+        }
+        /// <summary>
+        /// Flag for whether the node can be traversed or not
+        /// </summary>
+        public bool Walkable;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="pos">Position of node</param>
+        /// <param name="walkable">Can be traversed?</param>
+        /// <param name="weight">Weight of traversing</param>
+        public Node(Vector2 pos, bool walkable, float weight = 1)
+        {
+            Parent = null;
```

Comment:
Parent is not nullable, but we are assigning null to it here. If Parent needs to be null it should be marked as nullable at its declaration to address these warnings.